repository/src/test/java/org/alfresco/repo/serviceaccount/ServiceAccountRegistryImplTest.java

```diff
@@ -0,0 +1,157 @@
+        serviceAccountService.afterPropertiesSet();
+
+        Optional<String> testServiceAccount = serviceAccountService.getServiceAccountRole("testServiceAccount");
+        assertFalse("The service account role is not empty.", testServiceAccount.isEmpty());
+        assertEquals(ADMIN_SVC_AUTHORITY, testServiceAccount.get());
+        assertEquals(1, serviceAccountService.getServiceAccountNames().size());
+    }
+
+    @Test
+    public void testManyServiceAccounts() throws Exception
+    {
+        globalProperties.put("serviceaccount.role.testEditorSA", EDITOR_SVC_AUTHORITY);
+        globalProperties.put("serviceaccount.role.testCollaboratorSA", COLLABORATOR_SVC_AUTHORITY);
+        globalProperties.put("serviceaccount.role.testAdminSA", ADMIN_SVC_AUTHORITY);
+        serviceAccountService.afterPropertiesSet();
+
+        assertEquals(3, serviceAccountService.getServiceAccountNames().size());
+
+        Optional<String> editorSA = serviceAccountService.getServiceAccountRole("testEditorSA");
+        assertFalse("The service account role is not empty.", editorSA.isEmpty());
+        assertEquals(EDITOR_SVC_AUTHORITY, editorSA.get());
+
+        Optional<String> collaboratorSA = serviceAccountService.getServiceAccountRole("testCollaboratorSA");
+        assertFalse("The service account role is not empty.", collaboratorSA.isEmpty());
+        assertEquals(COLLABORATOR_SVC_AUTHORITY, collaboratorSA.get());
+
+        Optional<String> adminSA = serviceAccountService.getServiceAccountRole("testAdminSA");
+        assertFalse("The service account role is not empty.", adminSA.isEmpty());
+        assertEquals(ADMIN_SVC_AUTHORITY, adminSA.get());
+    }
+
+    @Test
+    public void testValidServiceAccountRoleValues() throws Exception
+    {
+        globalProperties.put("serviceaccount.role.testEditorSA", "EDITOR_SERVICE_ACCOUNT");
+        globalProperties.put("serviceaccount.role.testCollaboratorSA", COLLABORATOR_SVC_AUTHORITY);
+        globalProperties.put("serviceaccount.role.testAdminSA", "ADMIN_SERVICE_ACCOUNT");
+        serviceAccountService.afterPropertiesSet();
+
+        assertEquals(3, serviceAccountService.getServiceAccountNames().size());
+
+        Optional<String> editorSA = serviceAccountService.getServiceAccountRole("testEditorSA");
+        assertFalse("The service account role is not empty.", editorSA.isEmpty());
+        assertEquals(EDITOR_SVC_AUTHORITY, editorSA.get());
+
+        Optional<String> collaboratorSA = serviceAccountService.getServiceAccountRole("testCollaboratorSA");
+        assertFalse("The service account role is not empty.", collaboratorSA.isEmpty());
+        assertEquals(COLLABORATOR_SVC_AUTHORITY, collaboratorSA.get());
+
+        Optional<String> adminSA = serviceAccountService.getServiceAccountRole("testAdminSA");
+        assertFalse("The service account role is not empty.", adminSA.isEmpty());
+        assertEquals(ADMIN_SVC_AUTHORITY, adminSA.get());
+    }
```

Comment:
I can't spot the difference between these 2 tests cases, is there any? 🤔

Reply:
The first test is checking roles that start with the ROLE_ prefix, whereas the second test simulates the short form of role names, i.e., without the prefix. Thanks to your review, I've corrected a minor issue in the second test method.
The line `globalProperties.put("serviceaccount.role.testCollaboratorSA", COLLABORATOR_SVC_AUTHORITY)` should have been `globalProperties.put("serviceaccount.role.testCollaboratorSA", "COLLABORATOR_SERVICE_ACCOUNT")`.